docs/go/testing.md

```diff
@@ -245,3 +247,14 @@ func (s *UnitTestSuite) Test_ProgressWorkflow() {
 	s.Equal(value, 100)
 }
 ```
+
+## Debugging
+
+In addition to writing unit and integration tests, debugging your Workflows is also a very valuable testing tool.
+You can debug your Workflow code using a debugger provided by your favorite IDE.
+
+Note that when debugging your Workflow code, the Temporal Go SDK includes deadlock detection which fails a Workflow Task in case the code blocks over a second without relinquishing execution control.
+Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow code during debugging.
```

Comment:
```suggestion
Because of this you can often encounter a `PanicError: Potential deadlock detected` while stepping through Workflow Definitions during debugging.
```